saleor/dashboard/payment/forms.py

```diff
@@ -0,0 +1,60 @@
+from typing import Dict
+
+from django import forms
+from django.utils.text import slugify
+
+from ...extensions import ConfigurationTypeField
+from ...extensions.models import PluginConfiguration
+from ..forms import ConfigBooleanField, ConfigCharField
+
+TYPE_TO_FIELD = {
+    ConfigurationTypeField.STRING: ConfigCharField,
+    ConfigurationTypeField.BOOLEAN: ConfigBooleanField,
+}
+
+
+def create_custom_form_field(structure: Dict[str, str]) -> forms.Field:
+    elem_type = structure["type"]
+    return TYPE_TO_FIELD[elem_type](structure=structure)
+
+
+class GatewayConfigurationForm(forms.ModelForm):
+    class Meta:
+        model = PluginConfiguration
+        fields = ("active",)
+
+    def __init__(self, plugin, *args, **kwargs):
+        self.plugin = plugin
+        kwargs["instance"] = self._get_current_configuration()
+        super().__init__(*args, **kwargs)
+        # add new fields specified for gateway
+        self.fields.update(self._prepare_fields_for_given_config(self.instance))
+
+    def _get_current_configuration(self) -> PluginConfiguration:
+        qs = PluginConfiguration.objects.all()
+        return self.plugin.get_plugin_configuration(qs)
+
+    def _prepare_fields_for_given_config(
+        self, current_configuration: PluginConfiguration
+    ) -> Dict[str, forms.Field]:
+        parsed_fields = {}
+        structure = current_configuration.configuration
+        if structure is None:
+            raise Exception
```

Comment:
Well, `Exception` was a placeholder for `I will think about it later` - which I forgot to do :) IMO we can omit this if statement because the structure will be empty only when there is no default config and no structure defined in the plugin and this seems more like a feature than a bug.

Reply:
@korycins what do you think?

Reply:
I think that we can do nothing when configuriation is None. We have simillar flow for the Vatlayer nad Dashboard2.0. We don't have any fields in configuration we just show option to active/disable Vatlayer.